compiler/tools/fpp-to-cpp/test/component/base/ActiveGetProductsComponentAc.ref.cpp

```diff
@@ -142,17 +144,80 @@ Fw::SerializeStatus ActiveGetProductsComponentBase::DpContainer ::
   return status;
 }
 
+Fw::SerializeStatus ActiveGetProductsComponentBase::DpContainer ::
+  serializeRecord_StringArrayRecord(
+      const Fw::StringBase** array,
+      FwSizeType size
+  )
+{
+  FW_ASSERT(array != nullptr);
+  // Compute the size delta
+  const FwSizeType stringSize = 80;
+  FwSizeType sizeDelta =
+    sizeof(FwDpIdType) +
+    sizeof(FwSizeStoreType);
+  for (FwSizeType i = 0; i < size; i++) {
+    const Fw::StringBase *const sbPtr = array[i];
+    FW_ASSERT(sbPtr != nullptr);
+    sizeDelta += sbPtr->serializedTruncatedSize(stringSize);
+  }
+  // Serialize the elements if they will fit
+  Fw::SerializeStatus status = Fw::FW_SERIALIZE_OK;
+  if (this->m_dataBuffer.getBuffLength() + sizeDelta <= this->m_dataBuffer.getBuffCapacity()) {
```

Comment:
Prefer () instead of relying on order of operations:  `(this->m_dataBuffer.getBuffLength() + sizeDelta)`

Reply:
Good point! I will fix it.

Reply:
Fixed!